src/solc_0.7/common/BaseWithStorage/WithMetaTransaction.sol

```diff
@@ -6,44 +6,53 @@ import "./WithAdmin.sol";
-    }
-
-    /// @dev Enable or disable the ability of `metaTransactionProcessor` to perform meta-tx (metaTransactionProcessor rights).
-    /// @param metaTransactionProcessor address that will be given/removed metaTransactionProcessor rights.
-    /// @param processorType set the metaTransactionProcessor type
+    /// @dev Enable or disable the ability of metaTransactionProcessor.
+    /// to perform meta-tx (metaTransactionProcessor rights).
+    /// @param metaTransactionProcessor The address that will have metaTransactionProcessor rights
+    /// granted / revoked.
+    /// @param processorType The metaTransactionProcessor type to set.
     function setMetaTransactionProcessor(address metaTransactionProcessor, uint8 processorType) public {
         require(msg.sender == _admin, "ADMIN_ACCESS_DENIED");
         _setMetaTransactionProcessor(metaTransactionProcessor, processorType);
     }
 
+    /// @dev See setMetaTransactionProcessor
+    function _setMetaTransactionProcessor(address metaTransactionProcessor, uint8 processorType) internal {
+        _metaTransactionContracts[metaTransactionProcessor] = processorType;
+        emit MetaTransactionProcessor(metaTransactionProcessor, processorType);
+    }
+
+    /// @dev Check whether address `who` has been granted meta-transaction execution rights.
+    /// @param who The address to query.
+    /// @return The type of metatx processor (0 for none).
+    function getMetaTransactionProcessorType(address who) external view returns (uint8) {
+        return _metaTransactionContracts[who];
+    }
+
+    // --------------------------------------------------------------------------------
     // EIP-2771 Meta Transaction Recipient
+    // --------------------------------------------------------------------------------
+
+    /// @notice Check if forwarder is trusted.
+    /// @param forwarder The address to query.
+    /// @return whether or not forwarder is trusted.
     function isTrustedForwarder(address forwarder) public view returns (bool) {
         return _metaTransactionContracts[forwarder] == METATX_2771;
     }
 
-    function _setMetaTransactionProcessor(address metaTransactionProcessor, uint8 processorType) internal {
-        _metaTransactionContracts[metaTransactionProcessor] = processorType;
-        emit MetaTransactionProcessor(metaTransactionProcessor, processorType);
-    }
+    /// @dev Decide which sender address to use for this call.
+    /// If the call came through our trusted forwarder, return the original sender.
+    /// Otherwise, return `msg.sender`.
+    /// @return ret The sender of this call.
 
-    /**
-     * return the sender of this call.
-     * if the call came through our trusted forwarder, return the original sender.
-     * otherwise, return `msg.sender`.
-     * should be used in the contract anywhere instead of msg.sender
-     */
     function _msgSender() internal view virtual returns (address payable ret) {
```

Comment:
Wondered whether the name `ret` is intuitive enough

Reply:
good call, it's not very intuitive. Restored some additional comments.